### PR TITLE
config/application: improve status checks.

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
- Improve formatting by always adding a newline after each `raise` (as the current version of RuboCop recommends)
- Increase the timeout from 1 second to 5 seconds for all networked, GitHub API operations. As-is I could regularly reproduce timeouts by refreshing the `/_ping` endpoint ~10 times. This in turn results in flaky Pingdom checks which makes Classroom seem like it's down when it is not.
- Parse the new GitHub Status API and look specifically for the GitHub API as that's what we care about (e.g. not Issues and Pull Requests)
- Make the rate limit remaining considerable more conservative. Having 1,999 remaining requests should not make the service report a failing state.

CC @d12 and @johndbritton for review and thoughts.

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -76,24 +76,36 @@ class Application < Rails::Application
...
+
           "ok"
         end
 
-        ping.check :github do
-          uri = URI("https://status.github.com/api/status.json")
-          status = JSON.parse(Net::HTTP.get(uri))["status"]
-          raise "GitHub status is #{status}" unless status == "good"
+        ping.check :github, timeout: 5 do
+          status_components = JSON.parse(Net::HTTP.get(uri))["components"]
```
Comment (d12) [inline config/application.rb]: Where does `uri` come from here?

Comment (talsafran) [reply config/application.rb]: Good catch, I think it was accidentally removed from the diff.

```suggestion
          uri = URI("https://status.github.com/api/status.json")
          status_components = JSON.parse(Net::HTTP.get(uri))["components"]
```

Comment (MikeMcQuaid) [reply config/application.rb]: @d12 @talsafran Yup, missed on my `git add`! Adjusted to use the new GitHub Status API URI (`uri = URI("https://www.githubstatus.com/api/v2/components.json")`)

**Comment by MikeMcQuaid** [comment]
@d12 @BenEmdon @talsafran Oops, fixed now! Ready to 🚢 now, thanks!

File: config/application.rb [hunk 2]
```diff
@@ -76,24 +76,37 @@ class Application < Rails::Application
...
-          status = JSON.parse(Net::HTTP.get(uri))["status"]
-          raise "GitHub status is #{status}" unless status == "good"
+        ping.check :github, timeout: 5 do
+          uri = URI("https://www.githubstatus.com/api/v2/components.json")
+          status_components = JSON.parse(Net::HTTP.get(uri))["components"]
+          raise "GitHub status format is invalid!" if status_components.blank?
+
+          github_status_api_id = "brv1bkgrwx7q"
+          api_component = status_components.find do |component|
+            component["id"] == github_status_api_id
```
Comment (BenEmdon) [inline config/application.rb]: What is the id `brv1bkgrwx7q`? Is this global to all integrators? Is this a secret?

Comment (MikeMcQuaid) [reply config/application.rb]: It is the `id` for the GitHub Status API component from https://www.githubstatus.com/api/v2/components.json.

**Comment by MikeMcQuaid** [comment]
I don't have privileges to merge this. @talsafran @d12 @BenEmdon @tarebyte @srinjoym Could one of you who do do so? Thanks!

**Pull request merged**